### ISSUE TITLE
CA2000 - Exclude StringReader by default

### ISSUE THREAD

**Issue by ashmind** [dotnet/roslyn-analyzers#3212]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

[CA2000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2000)

#### Repro steps

```
var x = new StringReader("abc");
```

#### Expected behavior
`StringReader` is a well-known type that does not need to be disposed, so no warning.

#### Actual behavior
`CA2000` is reported.

**Comment by Evangelink** [comment]
That seems reasonable to me. We could either hardcode the exclusion or make this type part of the default value for the option (i.e. `dotnet_code_quality.CA2000.excluded_symbol_names = System.IO.StringReader`).

Tagging @mavasani

**Comment by Evangelink** [comment]
Ping @mavasani

**Comment by sharwell** [comment]
We can hard-code this exclusion. The lack of unmanaged resources is [documented for this type](https://docs.microsoft.com/en-us/dotnet/api/system.io.stringreader?view=netframework-4.8) and cannot be changed in the future.

**Comment by Evangelink** [comment]
@sharwell Any preference between excluding the type post-processing or adding the type directly in the dispose helper processing part? I would say the latter but this could have consequences on other rules so I am not sure.

**Comment by Evangelink** [comment]
Another interesting path could be to add this type to the list of `excluded_symbol_names` for the rule as we have allowed a custom exclusion for this kind of cases in #3085 .

**Comment by mavasani** [comment]
@Evangelink I think it is fine to add the exclusion in the dispose helper or the rule, whichever is convenient.
